src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -448,10 +452,26 @@ protected Task<ImmutableArray<CodeActionOperation>> PostProcessAsync(IEnumerable
     /// </summary>
     /// <param name="changedSolution">The solution changed by the <see cref="CodeAction"/>.</param>
     /// <param name="cancellationToken">A cancellation token</param>
+#pragma warning disable CA1822 // Mark members as static. This is a public API.
     protected Task<Solution> PostProcessChangesAsync(Solution changedSolution, CancellationToken cancellationToken)
-        => PostProcessChangesAsync(originalSolution: null!, changedSolution, cancellationToken);
+#pragma warning restore CA1822 // Mark members as static
+    => PostProcessChangesAsync(originalSolution: null!, changedSolution, cancellationToken);
 
-    internal async Task<Solution> PostProcessChangesAsync(
+    internal static async Task<ImmutableArray<(DocumentId documentId, CodeCleanupOptions codeCleanupOptions)>> GetDocumentIdsAndOptionsAsync(
+        Solution solution, CodeCleanupOptionsProvider optionsProvider, ImmutableArray<DocumentId> documentIds, CancellationToken cancellationToken)
```

Comment:
helper used here (for code actions that produce a solution (as opposed to a doc), and also in the FixAll code.